File: HideBloat/src/main/java/com/aliucord/plugins/hidebloat/widgets/SwitchItem.java
package com.aliucord.plugins.hidebloat.widgets;

import android.annotation.SuppressLint;
import android.content.Context;

import com.aliucord.Utils;
import com.aliucord.api.SettingsAPI;
import com.aliucord.utils.DimenUtils;
import com.aliucord.plugins.hidebloat.util.Const;
import com.discord.utilities.color.ColorCompat;
import com.discord.views.CheckedSetting;
import com.google.android.material.card.MaterialCardView;
import com.lytefast.flexinput.R;

@SuppressLint("ViewConstructor")
public class SwitchItem extends MaterialCardView {

    public final String viewName;

    public SwitchItem(Context context, SettingsAPI settingsAPI, String key, String viewName) {
        super(context);

        this.viewName = viewName;

        setRadius(DimenUtils.getDefaultCardRadius());
        setCardBackgroundColor(ColorCompat.getThemedColor(context, R.b.colorBackgroundSecondary));
        setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT));

        CheckedSetting setting = Utils.createCheckedSetting(context, CheckedSetting.ViewType.SWITCH, "Hide " + viewName, null);
        setting.setChecked(settingsAPI.getBool(key, Const.PREFERENCE_DEFAULT_VALUE));
        setting.setOnCheckedListener(v -> settingsAPI.setBool(key, v));

        addView(setting);
    }


}

File: plugins/LayoutController/src/main/java/com/xinto/aliuplugins/layoutcontroller/widgets/SwitchItem.kt
package com.xinto.aliuplugins.layoutcontroller.widgets

import android.annotation.SuppressLint
import android.content.Context
import android.widget.LinearLayout
import com.aliucord.Utils
import com.aliucord.api.SettingsAPI
import com.xinto.aliuplugins.layoutcontroller.util.PREFERENCE_DEFAULT_VALUE
import com.aliucord.utils.DimenUtils
import com.discord.utilities.color.ColorCompat
import com.discord.views.CheckedSetting
import com.google.android.material.card.MaterialCardView
import com.lytefast.flexinput.R
import com.xinto.aliuplugins.layoutcontroller.patchers.base.BasePatcher

@SuppressLint("ViewConstructor")
class SwitchItem(
    context: Context,
    settingsAPI: SettingsAPI,
    val patch: BasePatcher
) : MaterialCardView(context) {
    init {
        radius = DimenUtils.defaultPadding.toFloat()
        setCardBackgroundColor(ColorCompat.getThemedColor(context, R.b.colorBackgroundSecondary))
        layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT)

        Utils.createCheckedSetting(context, CheckedSetting.ViewType.SWITCH, patch.description, null)
            .apply {
                isChecked = settingsAPI.getBool(patch.key, PREFERENCE_DEFAULT_VALUE)
                setOnCheckedListener {
                    settingsAPI.setBool(patch.key, it)
                    if (patch.requiresRestart)
                        Utils.promptRestart()
                }
                this@SwitchItem.addView(this)
            }
    }
}
